gluten-core/src/main/scala/org/apache/gluten/extension/columnar/OffloadSingleNode.scala

```diff
@@ -299,16 +299,15 @@ case class OffloadProject() extends OffloadSingleNode with LogLevelUtil {
       // Project is still not transformable after remove `input_file_name` expressions.
       projectExec
     } else {
-      // the project with `input_file_name` expression should have at most
-      // one data source, reference:
+      // the project with `input_file_name` expression may have multiple data source
+      // by union all, reference:
       // https://github.com/apache/spark/blob/e459674127e7b21e2767cc62d10ea6f1f941936c
-      // /sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala#L506
+      // /sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala#L519
       val leafScans = findScanNodes(projectExec)
-      assert(leafScans.size <= 1)
-      if (leafScans.isEmpty || FallbackTags.nonEmpty(leafScans(0))) {
+      if (leafScans.isEmpty || leafScans.forall(FallbackTags.nonEmpty)) {
```

Comment:
looks for `union` the datatype and schema of al scans should be same. so it is ok to use `forall` here. thanks !

Reply:
I thought about it carefully and should use `exsits` here. Because fallback scan node cannot add `input_file_name` output. What do you think? @gaoyangxiaozhu @zhztheplayer

Reply:
hey @zml1206 ，using `exists` is better then `forall`, as `exists` from semantically, it more intuitively reflects that any we should fallback if any one scan node fallback. But here for `union` case, all scan has same schema,  that means if one scan node fallback, other scan nodes must also fallback, so i am fine here if still using `forall`.

But yes, changing to `exists` is better.